regex-syntax/src/ast/parse.rs

```diff
@@ -2435,6 +2522,29 @@ fn specialize_err<T>(
     }
 }
 
+/// Returns the span of the first capture group found. Returns None in case there are no capture groups.
+fn first_capture_group_span(ast: &Ast) -> Option<Span> {
```

Comment:
This might make parsing $\mathcal O(m^2)$, to investigate

Reply:
In case you only call the function once per look-around, and only search the span within the look-around, then it should be $`\mathcal{O}(m)`$. The only exception is when there are nested look-arounds. For this not to happen, you have to make sure not to induct on look-arounds inside this function (because these have already been checked not to contain capture groups.